<!DOCTYPE html>
<!--
    Copyright (C) 2017- Marcus Fedarko, Jay Ghurye, Todd Treangen, Mihai Pop
    Authored by Marcus Fedarko

    This file is part of MetagenomeScope.

    MetagenomeScope is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    MetagenomeScope is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with MetagenomeScope.  If not, see <http://www.gnu.org/licenses/>.
-->
<html lang="en">
    <head>
        <meta charset="utf-8" />
        <!-- Since we're using Bootstrap and jQuery here (and no other
             libraries that would conflict with old versions of IE), might as
             well keep this tag in to make this work a bit better with old
             versions of IE.
        -->
        <meta http-equiv="X-UA-Compatible" content="IE=edge" />
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <meta name="author" content="Marcus Fedarko" />
        <title>MetagenomeScope</title>
        <link rel="shortcut icon" href="res/icons/favicon.ico" />
        <link
            rel="stylesheet"
            href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css"
            integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u"
            crossorigin="anonymous"
        />
        <link rel="stylesheet" href="css/mgsc_docs_style.css" />
    </head>
    <body>
        <!-- CODELINK: Most of this markup defining this <nav> and its "collapse"
             button is adapted from Bootstrap's documentation on navbars,
             located here: https://getbootstrap.com/docs/3.3/components/#navbar
        -->
        <nav class="navbar navbar-default navbar-fixed-top">
            <div class="container-fluid">
                <!--
                    The navbar-header <div> contains stuff that'll be shown
                    when the navbar is collapsed (e.g. on small displays).
                -->
                <div class="navbar-header">
                    <button
                        type="button"
                        class="navbar-toggle collapsed"
                        data-toggle="collapse"
                        aria-expanded="false"
                        data-target="#main-navbar-content"
                    >
                        <span class="sr-only">Toggle navigation</span>
                        <!-- Each of these .icon-bar <spans> corresponds to a
                             line in the "hamburger" navigation element.
                        -->
                        <span class="icon-bar"></span>
                        <span class="icon-bar"></span>
                        <span class="icon-bar"></span>
                    </button>
                    <a class="navbar-brand" href="#">
                        MetagenomeScope
                    </a>
                </div>
                <!--
                    The navbar-collapse <div> contains stuff that'll be shown
                    when the navbar is uncollapsed.
                -->
                <div class="collapse navbar-collapse" id="main-navbar-content">
                    <ul class="nav navbar-nav">
                        <li><a href="#news">News</a></li>
                        <li><a href="#overview">Overview</a></li>
                        <!--
                        <li><a href="#demo">Demo</a></li>
                        <li><a href="#documentation">Documentation</a></li>
                        <li><a href="#installing">Installing</a></li>
                        -->
                        <li><a href="#code">Code</a></li>
                        <li><a href="#contact">Contact</a></li>
                    </ul>
                </div>
            </div>
        </nav>
        <!-- All <a> elements in #stuffbelowthenav will be underlined by
            default, contrary to Bootstrap's default. This makes these links
            easier to see for colorblind users (see css/mgsc_docs_style.css
            for details).
        -->
        <div id="stuffbelowthenav">
            <div class="page-header">
                <h1>MetagenomeScope</h1>
            </div>
            <div class="container-fluid">
                <div class="row">
                    <div class="col-lg-6">
                        <div class="thumbnail">
                            <img
                                onclick="openScreenshot(event);"
                                src="res/screenshots/marygold.png"
                                alt="Visualization of a graph based on Fig. 2(a) of Nijkamp et al. 2013, demonstrating MetagenomeScope's use of hierarchical pattern decomposition."
                            />
                            <hr />
                            <div class="caption">
                                <p>
                                    Visualization of a graph based on Fig. 2(a)
                                    of [<a
                                        href="https://www.ncbi.nlm.nih.gov/pubmed/24058058"
                                        >Nijkamp et al. 2013</a
                                    >].
                                </p>
                                <p>
                                    Due to MetagenomeScope's use of hierarchical
                                    pattern decomposition, the input assembly
                                    graph is automatically annotated as a bubble
                                    containing two smaller bubbles: each of the
                                    three bubbles can be interactively collapsed
                                    and uncollapsed in the interface.
                                </p>
                                <p>
                                    <strong
                                        >Check out a
                                        <a
                                            href="https://marbl.github.io/MetagenomeScope/demos/marygold"
                                            >demo</a
                                        >
                                        of this visualization!</strong
                                    >
                                </p>
                            </div>
                        </div>
                    </div>
                    <div class="col-lg-6">
                        <div class="thumbnail">
                            <img
                                onclick="openScreenshot(event);"
                                src="res/screenshots/srs049959_cc6.png"
                            />
                            <hr />
                            <div class="caption">
                                <p>
                                    Visualization of a region of a stool
                                    metagenome scaffold graph produced by
                                    <a
                                        href="https://github.com/marbl/MetaCarvel"
                                        >MetaCarvel</a
                                    >
                                    [<a
                                        href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1791-3"
                                        >Ghurye et al. 2019</a
                                    >].
                                </p>
                                <p>
                                    MetagenomeScope performs
                                    <a
                                        href="https://en.wikipedia.org/wiki/Layered_graph_drawing"
                                        >hierarchical graph layout</a
                                    >
                                    using
                                    <a href="https://graphviz.org/"
                                        >Graphviz'</a
                                    >
                                    <span class="dotname">dot</span> layout
                                    tool. This simplifies the visualization and
                                    manual inspection of complex, small-scale
                                    details within assembly graphs, such as
                                    these branching paths.
                                </p>
                            </div>
                        </div>
                    </div>
                </div>
            </div>

            <h2 id="news">News</h2>
            <p>
                <strong>December 2020:</strong> MetagenomeScope is back! We just
                merged in a new version of the codebase featuring many new
                features, including hierarchical pattern decomposition, bubble
                boundary duplication, drawing multiple components at once, and
                much more. Check out the code and installation instructions
                <a href="https://github.com/marbl/MetagenomeScope">on GitHub</a
                >!
            </p>
            <p>
                <strong>August 2018:</strong> Gave a presentation and tutorial
                on MetagenomeScope during the
                <a href="https://stamps.mbl.edu/index.php/Main_Page"
                    >2018 Strategies and Techniques for Analyzing Microbial
                    Population Structures (STAMPS) course</a
                >
                at the Marine Biological Laboratory! Check out the slides
                <a
                    href="https://drive.google.com/file/d/1Cyxi1AE_5n5EeYUHaeejp1BpyOAVAr_V/view"
                    >here</a
                >, and check out the command-line MetagenomeScope tutorial
                <a
                    href="https://github.com/marbl/MetagenomeScope/blob/master/STAMPS_tutorial.md"
                    >here</a
                >.
            </p>
            <p>
                <strong>January 2018:</strong>
                Presented a poster on MetagenomeScope at the
                <a
                    href="https://blog.umd.edu/m3/upcoming-events/2018-winter-mid-atlantic-microbiome-meet-up/"
                    >2018 Winter Mid-Atlantic Microbiome Meetup
                    (M<sup>3</sup>)</a
                >!
            </p>
            <p>
                <strong>September 2017:</strong>
                Presented a poster on MetagenomeScope at
                <a href="https://gd2017.ccis.northeastern.edu/">GD 2017</a>!
                Check out the most recent version of the poster
                <a href="res/gd2017_mgsc_poster_updated.pdf">here</a>.
            </p>

            <h2 id="overview">Overview</h2>
            <p>
                MetagenomeScope is a web-based visualization tool designed for
                metagenomic assembly graphs. MetagenomeScope aims to display a
                semi-linearized, hierarchical overview of the input graph while
                emphasizing the presence of certain structural patterns in the
                graph.
            </p>
            <p>
                The tool consists of a preprocessing script (written in Python)
                and a visualization interface (a client-side web application).
                The preprocessing script takes as input an assembly graph file
                (currently LastGraph, GFA, FASTG, and
                <a href="https://github.com/marbl/MetaCarvel">MetaCarvel</a> GML
                files are accepted) and creates a directory containing the HTML
                / JS / CSS visualization.
            </p>
            <p>
                MetagenomeScope is still under active development, so please
                <a href="#contact">get in touch with us</a> if you have any
                questions or other feedback about the tool.
            </p>
            <h3>Highlighted features</h3>
            <ul class="highlighted-features">
                <li>
                    <strong>Hierarchical graph layout</strong> performed using
                    <a href="https://www.graphviz.org">Graphviz</a>'
                    <span class="dotname">dot</span> layout engine.
                </li>
                <li>
                    <strong>Hierarchical pattern decomposition:</strong>
                    identifies and supports interactive collapsing/uncollapsing
                    of bubbles and frayed ropes [<a
                        href="https://www.ncbi.nlm.nih.gov/pubmed/20211242"
                        >Miller, Koren, Sutton 2010</a
                    >], along with linear and cyclic linear sequences of
                    contigs.
                    <ul>
                        <li>
                            Runs pattern decomposition iteratively, so that
                            patterns containing other patterns (e.g. bubbles
                            containing chains) can be identified. In the viewer
                            interface, these patterns can be collapsed or
                            uncollapsed to show the graph at different levels of
                            "complexity."
                        </li>
                    </ul>
                </li>
                <!--
                <li>
                    Supports the
                    <strong
                        >visualization of
                        <a
                            href="https://www.ncbi.nlm.nih.gov/assembly/agp/AGP_Specification/"
                            >AGP files</a
                        ></strong
                    >
                    describing scaffolds constructed for the graph
                </li>
                <li>
                    Includes controls for
                    <strong>manually performing finishing</strong> on sequences
                    of contigs in the graph
                    <ul>
                        <li>
                            Supports "autofinishing," in which unambiguous paths
                            are automatically followed during this process
                        </li>
                        <li>
                            The resulting paths can be exported to
                            <a
                                href="https://www.ncbi.nlm.nih.gov/assembly/agp/AGP_Specification/"
                                >AGP</a
                            >
                            or
                            <a
                                href="https://en.wikipedia.org/wiki/Comma-separated_values"
                                >CSV</a
                            >
                            format
                        </li>
                    </ul>
                </li>
                -->
                <li>
                    <strong>Scales edges</strong> in the graph by their
                    multiplicity/bundle size [<a
                        href="https://www.ncbi.nlm.nih.gov/pubmed/20211242"
                        >Miller, Koren, Sutton 2010</a
                    >].
                    <ul>
                        <li>
                            Uses
                            <a
                                href="http://sphweb.bumc.bu.edu/otlt/mph-modules/bs/bs704_summarizingdata/bs704_summarizingdata7.html"
                                >Tukey fences</a
                            >
                            to identify "outlier" edge weights and omit them
                            from the scaling process
                        </li>
                        <li>
                            Outlier edges are specially colored in the graph to
                            indicate their unusual high/low weight
                        </li>
                    </ul>
                </li>
                <!--
                <li>
                    <strong>Colorblind support:</strong> every color used in the
                    graph display of the application is easily configurable
                    <ul>
                        <li>
                            Color settings can be exported/imported as
                            <a
                                href="https://en.wikipedia.org/wiki/Tab-separated_values"
                                >TSV</a
                            >
                            files
                        </li>
                    </ul>
                </li>
                <li>
                    Supports node colorization by
                    <a href="https://en.wikipedia.org/wiki/GC-content"
                        >GC content</a
                    >
                    and repeat annotations; more to come soon
                </li>
                -->
            </ul>
            <!--
            <h2 id="demo">Demo</h2>
            <p>
                A demo of MetagenomeScope's viewer interface (with about a dozen
                sample database files available for use) is available online at
                <a href="https://mgsc.umiacs.io">mgsc.umiacs.io</a>. You can
                also upload your own database files generated by the
                preprocessing script to visualize them in the tool.
            </p>
            <h2 id="documentation">Documentation</h2>
            <p>
                Documentation about using MetagenomeScope is available in its
                GitHub wiki, located
                <a href="https://github.com/marbl/MetagenomeScope/wiki">here</a
                >. The wiki is a work in progress, so if you have any further
                questions please feel free to <a href="#contact">contact us</a>.
            </p>
            <h2 id="installing">Installing</h2>
            <p>
                Instructions for installing MetagenomeScope's preprocessing
                script are available on the wiki
                <a
                    href="https://github.com/marbl/MetagenomeScope/wiki/Installation-Instructions"
                    >here</a
                >.
            </p>
            <p>
                You can visualize the database files generated by the
                preprocessing script in MetagenomeScope's viewer interface,
                either on the <a href="https://mgsc.umiacs.io">demo</a> or in a
                local copy of the viewer interface file (see
                <a
                    href="https://github.com/marbl/MetagenomeScope/wiki/Viewer-Interface-Overview"
                    >this page</a
                >
                of the wiki for more information).
            </p>
            <p>
                If you want, you can also create and host your own demos of
                MetagenomeScope's viewer interface to share database files with
                the public: see
                <a
                    href="https://github.com/marbl/MetagenomeScope/wiki/Customizing-Your-Own-Demo"
                    >this page</a
                >
                of MetagenomeScope's wiki for details.
            </p>
            -->
            <h2 id="code">Code</h2>
            <p>
                The source code for MetagenomeScope is publicly available on
                GitHub
                <a href="https://www.github.com/marbl/MetagenomeScope"> here</a
                >. MetagenomeScope is licensed under the
                <a href="https://www.gnu.org/licenses/gpl-3.0.en.html">
                    GNU General Public License, version 3</a
                >. (If you're interested, the source code for this site is
                <a
                    href="https://www.github.com/marbl/MetagenomeScope/tree/master/docs"
                >
                    also available on GitHub</a
                >.)
            </p>
            <p>
                See the
                <a
                    href="https://github.com/marbl/MetagenomeScope/wiki/Acknowledgements"
                    >acknowledgements</a
                >
                section of MetagenomeScope's wiki for an up-to-date list of the
                third-party libraries used by MetagenomeScope.
            </p>
            <h2 id="contact">Contact</h2>
            <p>
                MetagenomeScope was created by members of the
                <a href="https://sites.google.com/a/cs.umd.edu/poplab/"
                    >Pop Lab</a
                >
                in the
                <a href="https://cbcb.umd.edu"
                    >Center for Bioinformatics and Computational Biology</a
                >
                at the
                <a href="https://umd.edu"
                    >University of Maryland, College Park</a
                >.
            </p>
            <p>
                Feel free to email <code>mfedarko (at) umd (dot) edu</code> with
                any questions, suggestions, comments, concerns, etc. regarding
                the tool. If you'd like, you can also
                <a href="https://github.com/marbl/MetagenomeScope/issues/"
                    >open an issue</a
                >
                on MetagenomeScope's GitHub page.
            </p>
        </div>
        <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
        <script
            src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"
            integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa"
            crossorigin="anonymous"
        ></script>
        <script src="js/extra_functionality.js"></script>
    </body>
</html>
